### ISSUE TITLE
[CI Failure]:  mi355_1: V1 Sample + Logits

### ISSUE THREAD

**Issue by AndreasKaratzas** [vllm-project/vllm#41336]
### Name of failing test

`(command rocm-smi || true) && export VLLM_TEST_GROUP_NAME=mi355_1-v1-sample---logits && export VLLM_ALLOW_DEPRECATED_BEAM_SEARCH=1 && cd /vllm-workspace/tests && pytest -v -s v1/sample && pytest -v -s v1/logits_processors && pytest -v -s v1/test_oracle.py && pytest -v -s v1/test_request.py && pytest -v -s v1/test_outputs.py`

### Basic information

- [ ] Flaky test
- [x] Can reproduce locally
- [ ] Caused by external libraries (e.g. bug in `transformers`)

### 🧪 Describe the failing test

```
FAILED v1/sample/test_logprobs.py::test_spec_decode_logprobs[ngram-raw_logprobs]
FAILED v1/sample/test_logprobs.py::test_spec_decode_logprobs[ngram-processed_logprobs]
```

### 📝 History of failing test

- Break frequency (60d, pass↔fail flips): 0
- Latest nightly date: 2026-04-29
- Latest build(s): [amd-ci #8058](https://buildkite.com/vllm/amd-ci/builds/8058)
- Latest hardware status: `mi355_1`=fail

**Comment by AndreasKaratzas** [comment]
@AndreasKaratzas is triaging this TG.

**Comment by AndreasKaratzas** [comment]
There is a PR up for this TG:
- https://github.com/vllm-project/vllm/pull/41335
